example/prometheus/main.go

```diff
@@ -58,13 +60,13 @@ func main() {
 	}
 
 	// This is the equivalent of prometheus.NewCounterVec
-	counter, err := meter.SyncFloat64().Counter("foo", instrument.WithDescription("a simple counter"))
+	counter, err := meter.SyncFloat64().Counter("foo", syncfloat64.WithDescription("a simple counter"))
```

Comment:
Agreed, this is not ideal.

Though, introducing `instrument.WithInt64Callback` will mean need it needs to return an option that only the `asyncint64` instruments will accept. Meaning,

```go
func WithInt64Callback(callback /* ... */) Int64OberserverOption
```

Similarly for the `float64` case.

This means that the `instrument` package will need to add both `Int64OberserverOption` and `Float64OberserverOption`, `Int64OberserverConfig` and `Float64OberserverConfig`, `NewInt64OberserverConfig` and `NewFloat64OberserverConfig`, and the `Int64Observer` and `Float64Observer` types to be used in the callbacks.  This will allow then for the existing `instrument` description and unit options to be implementable by options that satisfy the existing `Config` and the added `Int64OberserverConfig` and `Float64OberserverConfig` types.

So adding the option at this level greatly complicates the `instrument` package.

It also seems fair to ask at that point, why not also provide separate options for the sync instruments as well to ensure forward compatibility when they need distinct options of their own. Which will only further add to the jumble.

I'm not opposed to the approach, as I see the need to import options from another package as problematic, but I'm also worried about the explosion of types making things hard to understand how it is used.

Thoughts?

Reply:
Do the config types need to be exported?

Reply:
I do still prefer having the options in the instrumentation package.  We are adding a large number of types and options in the current version as well, albeit split into smaller packages.

Reply:
> Do the config types need to be exported?

Yes, so the SDK can process the options into a unified config they can interpret in their implementations.

Reply:
> I do still prefer having the options in the instrumentation package. We are adding a large number of types and options in the current version as well, albeit split into smaller packages.

Okay, I'll try updating later today or early next week. :+1:

Reply:
Updated, PTAL.